src/trexio_tools/trexio_run.py

```
@@ -46,6 +46,11 @@ def main(filename=None, args=None):
     else:
         back_end = trexio.TREXIO_HDF5
 
+    if args["--motype"] is not None:
+        motype = str(args["--motype"]).upper()
+    else:
+        motype = 'NATURAL'
```

We need to be careful here because if the MOs are not consistent with the determinants, the file will be all wrong. We should let resultsFile decide here which MOs are the correct ones if `--motype` is `None`. You can leave `motype=None` and let it propagate to the `run_resultsFile` function.

---

Change the above portion to

```
   if args["--motype"] is not None:
        motype = str(args["--motype"])
    else:
        motype = None
```

---

We can safely remove the aforementioned code. You can use `motype=args["--motype"]` in the arguments list of the `run` function, i.e.
```
    elif args["convert-from"]:
        from converters.convert_from import run
        run(args["TREXIO_FILE"], args["--input"], args["--type"], back_end=back_end, motype=args["--motype"])
```

---

sounds good to me.

---

You are right, my bad!

---

@neelravi I just checked and my proposed change works fine. This is because if the `motype` is not specified in the command line then `args["--motype"] is None`, which is what you use in your `if else`. So this can go directly in the arguments list :100:

---

Awesome! I will patch it.